### ISSUE TITLE
[BUG] Quarterly Triangles Not Forming Properly

### ISSUE THREAD

**Issue by mlockwo1** [casact/chainladder-python#431]
**Describe the bug**
When creating triangles with annual origin periods and quarterly development periods, the development period labels don't consistently populate correctly by incrementing by 3s (e.g. 18, 21, 24, 26, 29...). I believe this creates problems down the road when using these triangles for development in that the development doesn't calculate on origin years that are long enough to have these oddball development periods.

[quarterly.xlsx](https://github.com/casact/chainladder-python/files/11435621/quarterly.xlsx)

**To Reproduce**
```python
import pandas as pd
import numpy as np
import chainladder as cl

#I copy/pasted the quarterly CSV from git to an excel file and saved on my desktop. Load the quarterly data as you see fit.

quarterly_data = pd.read_excel('C:\\...Desktop\\quarterly.xlsx')
quarterly = cl.Triangle(
    loss_data,
    origin = 'origin',
    development='development',
    columns='incurred',
    cumulative=True
)
quarterly
```

Note the time periods on the triangle don't always increment by 3.

```python
quarterly_dev = cl.Pipeline(
    [("dev", cl.Development()), ("tail", cl.TailCurve())]
).fit_transform(quarterly)

quarterly_model = cl.Chainladder().fit(quarterly_dev)
quarterly_model.ultimate_
```
Only ultimates for the latest 2 periods, which aren't impacted by the oddball time periods, are created.

**Expected behavior**
The triangle should increment by 3s along the development months, and in so doing create ultimates for all origin periods present.

**Desktop (please complete the following information):**
pandas: 2.0.1
numpy: 1.23.5
chainladder: 0.8.15

**Comment by jbogaardt** [comment]
Hello @mlockwo1, welcome! Thank you for the reproducible example. I'm having a hard time getting the buggy answer. I see ultimates for all origins using your code:
![image](https://github.com/casact/chainladder-python/assets/19401133/33ef2bd2-d397-4d1e-8e84-d2d2660d0b93)
To be clear, you don't see the same output?

**Comment by mlockwo1** [comment]
UPDATE:
I reverted my pandas package back to 1.5.3 as that's what you used in the onboarding tutorial. That seems to have fixed the issue. I can not get my data to work properly. It's possible that the updates to pandas somehow broke that functionality.

I appreciate the help, looking forward to using his package!
